pyresample/bucket/__init__.py

```diff
@@ -228,8 +237,11 @@ def get_sum(self, data, skipna=True):
             data = data.data
         data = data.ravel()
 
-        # Remove NaN values from the data when used as weights
-        weights = da.where(np.isnan(data), 0, data)
+        # Remove fill_values values from the data when used as weights
+        if np.isnan(fill_value):
+            weights = da.where(np.isnan(data), 0, data)
+        else:
+            weights = da.where(data == fill_value, 0, data)
```

Comment:
Instead of the `if` statement, what about a helper for getting the invalid mask that does the nan check on the fill value and returns the appropriate mask? That way this stays `weights = da.where(invalid_mask, 0, data)`.

Reply:
Done